ceno_zkvm/src/instructions/riscv/jump/auipc.rs

```diff
@@ -74,10 +76,10 @@ impl<E: ExtensionField> Instruction<E> for AuipcInstruction<E> {
         step: &ceno_emul::StepRecord,
     ) -> Result<(), ZKVMError> {
         let pc: u32 = step.pc().before.0;
-        let imm: u32 = step.insn().imm_internal();
-        let (sum, overflow) = pc.overflowing_add(imm);
+        let imm = InsnRecord::imm_internal(&step.insn());
+        let (sum, overflow) = pc.overflowing_add(imm as u32);
 
-        set_val!(instance, config.imm, imm as u64);
+        set_val!(instance, config.imm, i64_to_base::<E::BaseField>(imm));
```

Comment:
It's a bit of a shame that we need to add `<E::BaseField>` everywhere now.  I wonder if we can avoid that.

Reply:
What about a From<i64>?

Reply:
That would be best, but to make that work, you need to make a change to the crate that implements Goldilocks. (Or you need to introduce an extra wrapper.)

Rust doesn't allow 'orphans'.